drivers/wifi/esp/esp.c

```diff
@@ -722,6 +722,21 @@ static int esp_mgmt_ap_disable(const struct device *dev)
 	return ret;
 }
 
+static void esp_configure_hostname(struct esp_data *data)
+{
+#if defined(CONFIG_NET_HOSTNAME_ENABLE)
+	char cmd[64];
```

Comment:
I am not fond of these hard coded values, what about putting here
```
char cmd[sizeof("AT+CWHOSTNAME=''") +
         sizeof(CONFIG_NET_HOSTNAME) +
         IS_ENABLED(CONFIG_NET_HOSTNAME_UNIQUE) ? sizeof("0011223344556677") : 0);
```

Reply:
Then what about adding this to `net/hostname.h`:
```
#if defined(CONFIG_NET_HOSTNAME_ENABLE)
#if defined(CONFIG_NET_HOSTNAME_UNIQUE)
/* Allocate extra space to append MAC address to hostname */
#define NET_HOSTNAME_EXTRA_SPACE (8 * 2)
#else
#define NET_HOSTNAME_EXTRA_SPACE 0
#endif /* CONFIG_NET_HOSTNAME_UNIQUE */

#define NET_HOSTNAME_MAX_SIZE	(sizeof(CONFIG_NET_HOSTNAME) +	\
				 NET_HOSTNAME_EXTRA_SPACE)
#endif /* CONFIG_NET_HOSTNAME_ENABLE */
```
or alternatively (based on your above comment):
```
#if defined(CONFIG_NET_HOSTNAME_ENABLE)
#define NET_HOSTNAME_MAX_SIZE	(sizeof(CONFIG_NET_HOSTNAME) +	\
				 IS_ENABLED(CONFIG_NET_HOSTNAME_UNIQUE) ? \
				 sizeof("0011223344556677") : 0)
#endif /* CONFIG_NET_HOSTNAME_ENABLE */
```
and updating `hostname.c` with:
```
static char hostname[NET_HOSTNAME_MAX_SIZE];
```
Then in ESP we can use:
```
char cmd[sizeof("AT+CWHOSTNAME=''") + NET_HOSTNAME_MAX_SIZE];
```

Reply:
The 2nd alternative looks better, +1 to that.